R/utils.R

```diff
@@ -1161,10 +1161,31 @@ countAndPrint <- function(x) {
     } else if (grepl("\\.cdf($|\\.)|\\.nc($|\\.)", x, ignore.case = TRUE)) {
         return("netCDF")
     } else {
-        stop("Could not determine file type for ", x)
+        return(.mzRBackendFromContent(x))
     }
 }
 
+#' Determine the backend from the (first few lines of the) file content.
+#' 
+#' @author Johannes Rainer
+#'
+#' @noRd
+.mzRBackendFromContent <- function(x = character()) {
+    if (length(x) != 1)
+        stop("parameter 'x' has to be of length 1")
+    suppressWarnings(
+        first_lines <- readLines(x, n = 4)
+    )
+    if (any(grepl("<mzML", first_lines)) | any(grepl("<mzXML", first_lines))) {
```

Comment:
I would suggest `any(grepl("<mz[X]?ML", first_lines)` here to avoid two `grepl` calls. 

BTW: if two `grepl` calls are needed I would also suggest `||` to avoid the second call if the first is already `TRUE`.